plugins/resteasy-http/src/main/java/org/radargun/service/RESTEasyHTTPOperations.java

```diff
@@ -0,0 +1,226 @@
+            } catch (Exception e) {
+               throw new RuntimeException("RESTEasyHTTPOperations::get request threw exception: " + target, e);
+            }
+         }
+         return result;
+      }
+
+      @Override
+      public boolean containsKey(K key) {
+         if (service.isRunning()) {
+            return get(key) != null;
+         }
+         return false;
+      }
+
+      @Override
+      public void put(K key, V value) {
+         if (service.isRunning()) {
+            try {
+               String target = buildUrl(service.cacheName, key);
+               Invocation put = service.getHttpClient().target(target).request().accept(service.getContentType())
+                     .buildPut(Entity.entity(encodeObject(value), service.getContentType()));
+               Response response = put.invoke();
+               int status = response.getStatus();
+               response.close();
+               if (status != Status.OK.getStatusCode()) {
+                  throw new RuntimeException("RESTEasyHTTPOperations.put::Unexpected HttpStatus: " + status + " "
+                        + response.getStatusInfo().getReasonPhrase() + " for request " + target);
+               }
+            } catch (IOException e) {
+               throw new RuntimeException(e);
+            } catch (NullPointerException e) {
+               throw new RuntimeException(e);
+            } catch (IllegalArgumentException e) {
+               throw new RuntimeException(e);
+            }
+         }
+      }
+
+      @Override
+      public V getAndPut(K key, V value) {
+         // Not Atomic
+         V prevValue = null;
+         if (service.isRunning()) {
+            if (containsKey(key)) {
+               prevValue = get(key);
+            }
+            put(key, value);
+         }
+         return prevValue;
+      }
+
+      @Override
+      public boolean remove(K key) {
+         if (service.isRunning()) {
+            String target = buildUrl(service.cacheName, key);
+            Invocation delete = service.getHttpClient().target(target).request().accept(service.getContentType())
+                  .buildDelete();
+            Response response = delete.invoke();
+            int status = response.getStatus();
+            response.close();
+            if (status != Status.OK.getStatusCode()) {
+               if (status == Status.NOT_FOUND.getStatusCode()) {
+                  log.warn("RESTEasyHTTPOperations.remove::Key: " + key + " does not exist in cache: "
+                        + service.cacheName);
+               } else {
+                  throw new RuntimeException("RESTEasyHTTPOperations.remove::Unexpected HttpStatus: " + status + " "
+                        + response.getStatusInfo().getReasonPhrase() + " for request " + target);
+               }
+            }
+            return true;
+         }
+         return false;
+      }
+
+      @Override
+      public V getAndRemove(K key) {
+         // Not Atomic
```

Comment:
Thanks for posting this doc, @rvansa ! I'll read through to make sure I understand your suggestion here.